Source/DataAccessLayer/Utility.cst

```diff
@@ -836,9 +844,8 @@ private void WriteExecuteMethod(ExecuteType executeType, bool useTransactions)
                     RetryMaxAttempts,
                     sleepType,
                     "DataSet",
-                    (useTransactions? ", transactionManager.TransactionObject":""),
-                    (useTransactions? "transactionManager.Database":"database"),
-                    (useTransactions? "if (!transactionManager.IsOpen) throw new DataException(\"Transaction must be open before executing a query.\");":""),
+                    string.Format(returnResults, "DataSet"),
+                    (useTransactions? "if (!transactionManager.IsOpen && !transactionManager.IsDisabled) throw new DataException(\"Transaction must be open before executing a query.\");":""),

```

Comment:
Can we break these into variables so it's easier to read what's being passed into the format.